source/kraken/realtime.cpp

```diff
@@ -225,10 +245,10 @@ static bool is_added_service(const transit_realtime::TripUpdate& trip_update) {
                 && trip_update.stop_time_update_size()) {
         for (const auto& st: trip_update.stop_time_update()) {
             // adding a stop_time event (adding departure or/and arrival) is adding service
-            if (get_relationship(st.departure(), st) ==
-                        trt::TripUpdate_StopTimeUpdate_ScheduleRelationship_ADDED
-                    || get_relationship(st.arrival(), st) ==
-                            trt::TripUpdate_StopTimeUpdate_ScheduleRelationship_ADDED) {
+            if (get_status(st.departure(), st) == StopTimeUpdate::Status::ADDED ||
```

Comment:
Can you not re-use your function from `pb_convertre.cpp` ?
```cpp
bool is_detour(nd::StopTimeUpdate::Status dep_status, nd::StopTimeUpdate::Status arr_status) const
```
If you do, you might want to move the function on a better suited place ;)